### PR TITLE
MS-36 MacOS - Create product list view

### PR DESCRIPTION
**Pull request by DianaZeledon**
Screenshot 2024-12-14 at 8 55 17 PM

---

### **Type of Change**

- [ ] Bug fix
- [X] New feature
- [ ] Repository settings
- [ ] Documentation updated

---

### CODE REVIEW

**Review by Rexmoon — CHANGES_REQUESTED** [review]
Hey @DianaZeledon,

I noticed some issues comparing with figma design:

- Row should has background color (White and dark as figma)
- Product's columns is not aligment
- Product's picture is in the same column as product name
- Product's name allows more than 1 line
- Category's badge has black text color
- Category's green badge is darker as figma
- Edit and delete icons don't match
- Text fonts don't match
- Pagination row is missed (Mocked)

Take a look please.

Screenshot 2024-12-15 at 12 36 45

**Comment by DianaZeledon** [comment]
> Hey @DianaZeledon,
> 
> I noticed some issues comparing with figma design:
> 
> * Row should has background color (White and dark as figma)
> * Product's columns is not aligment
> * Product's picture is in the same column as product name
> * Product's name allows more than 1 line
> * Category's badge has black text color
> * Category's green badge is darker as figma
> * Edit and delete icons don't match
> * Text fonts don't match
> * Pagination row is missed (Mocked)
> 
> Take a look please.
> 
> Screenshot 2024-12-15 at 12 36 45

I have tried to replicate as much as I can, let me know if more changes are needed.
image

File: moon-store-mac-os/moon-store-mac-os/Helpers/Components/SearchView.swift [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+                if isSearchFocused, !searchText.isEmpty {
+                    clearButton
+                }
+            }
+
+            if isSearchFocused {
+                cancelButton
+            }
+        }
+        .frame(width: Constants.searchWidht)
```
Comment (Rexmoon) [inline moon-store-mac-os/moon-store-mac-os/Helpers/Components/SearchView.swift]:
```suggestion
        .frame(width: Constants.searchWidth)
```

File: moon-store-mac-os/moon-store-mac-os/Views/Product/ProductListView.swift [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+    case options
+    
+    var title: String {
+        switch self {
+            case .product: "Producto"
+            case .category: "Categoría"
+            case .inStock: "En Stock"
+            case .price: "Precio"
+            case .options: ""
+            case .productIcon: ""
```
Comment (Rexmoon) [inline moon-store-mac-os/moon-store-mac-os/Views/Product/ProductListView.swift]:
```suggestion
            case .options, .productIcon: ""
```

File: moon-store-mac-os/moon-store-mac-os/Views/Product/ProductListView.swift [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+                } label: {
+                    Label(Constants.Button.title, systemImage: Constants.plusIcon)
+                        .foregroundStyle(.msWhite)
+                }
+                .buttonStyle(.plain)
+                .padding(Constants.cornerRadius)
+                .background {
+                    RoundedRectangle(cornerRadius: Constants.cornerRadius)
+                        .fill(.msPrimary)
+                }
```
Comment (Rexmoon) [inline moon-store-mac-os/moon-store-mac-os/Views/Product/ProductListView.swift]: You could use 
```
.background(.msPrimary, in: ...)
```

File: moon-store-mac-os/moon-store-mac-os/Views/Product/ProductMockData.swift [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    case otros = "Otros"
+    
+    // Color asociado a cada categoría
+    var color: Color {
+        switch self {
+        case .frutas: return .green
+        case .lacteos: return .blue
+        case .panaderia: return .orange
+        case .verduras: return .red
+        case .otros: return .gray
```
Comment (Rexmoon) [inline moon-store-mac-os/moon-store-mac-os/Views/Product/ProductMockData.swift]: You could use implicit return.
```suggestion
        case .frutas: .green
        case .lacteos: .blue
        case .panaderia: .orange
        case .verduras: .red
        case .otros: .gray
```

File: moon-store-mac-os/moon-store-mac-os/Views/Product/ProductMockData.swift [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+enum CategoryType: String, CaseIterable {
+    case frutas = "Frutas"
+    case lacteos = "Lácteos"
+    case panaderia = "Panadería"
+    case verduras = "Verduras"
+    case otros = "Otros"
+    
+    // Color asociado a cada categoría
```
Comment (Rexmoon) [inline moon-store-mac-os/moon-store-mac-os/Views/Product/ProductMockData.swift]: Code documentation in English please.

File: moon-store-mac-os/moon-store-mac-os/Views/Product/ProductMockData.swift [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+        switch self {
+        case .frutas: return .green
+        case .lacteos: return .blue
+        case .panaderia: return .orange
+        case .verduras: return .red
+        case .otros: return .gray
+        }
+    }
+}
+
```
Comment (Rexmoon) [inline moon-store-mac-os/moon-store-mac-os/Views/Product/ProductMockData.swift:37]: Add MARK

**Review by Rexmoon — CHANGES_REQUESTED** [review]
Little comments please take a look. Thanks

**Review by Rexmoon — APPROVED** [review]
Amazing!

@DianaZeledon please merge 👍

**Pull request merged**